requre/storage.py

```diff
@@ -86,6 +87,71 @@ class DataTypes(Enum):
     Dict = 2
 
 
+class StorageKeyStrategy(Enum):
+    Full = 1
+    Simple = 2
+    SuperSimple = 3
+    Custom = 4
+    Default = Full
+
+    @staticmethod
+    def __get_base_keys_custom(func: Callable) -> List[Any]:
+        raise NotImplementedError(
+            "You have to redefine this method to use custom store format"
+        )
+
+    @staticmethod
+    def __get_base_keys_full(func: Callable) -> List[Any]:
+        output: List[str] = list()
+        # callers module list, to be able to separate requests for various services in one file
+        caller_list: List[str] = list()
+        for currnetframe in inspect.stack():
+            if inspect.getmodule(currnetframe[0]):
+                module_name = inspect.getmodule(currnetframe[0]).__name__
+                # avoid to store requre.storage to module stack
+                # backward compatibility issue
+                if module_name.startswith("requre.storage"):
+                    continue
+            else:
+                continue
+            if module_name.startswith("_"):
+                break
+            else:
+                if len(caller_list) and caller_list[-1] == module_name:
+                    continue
+                else:
+                    caller_list.append(module_name)
+        output += caller_list[::-1]
+        # module name where function is
+        output.append(inspect.getmodule(func).__name__)
+        # name of function what were used
+        output.append(func.__name__)
+        return output
+
+    @staticmethod
+    def __get_base_keys_simple(func: Callable) -> List[Any]:
+        return [inspect.getmodule(func).__name__, func.__name__]
+
+    @staticmethod
+    def __get_base_keys_super_simple(func: Callable) -> List[Any]:
+        return [func.__name__]
+
+    @classmethod
+    def get_base_keys(cls, func: Callable) -> List[Any]:
+        if DataMiner().storage_key_strategy == cls.SuperSimple:
+            return StorageKeyStrategy.__get_base_keys_super_simple(func)
+        elif DataMiner().storage_key_strategy == cls.Simple:
+            return StorageKeyStrategy.__get_base_keys_simple(func)
+        elif DataMiner().storage_key_strategy == cls.Full:
+            return StorageKeyStrategy.__get_base_keys_full(func)
+        elif DataMiner().storage_key_strategy == cls.Custom:
+            return StorageKeyStrategy.__get_base_keys_custom(func)
+        else:
```

Comment:
Is it possible to have multiple classes with the `get_base_keys` implementation?

Reply:
@lachmanfrantisek sure, but not sure if it is a good idea, because then I need some function or something that decide which class to use.